### PR TITLE
Post sphinx-clean up and contributing guide updates

### LINKED ISSUE #14762: make docs failing
https://github.com/huggingface/transformers/blob/2a606f9974feb0f7578e6a638c7e5b548523ecb4/CONTRIBUTING.md?plain=1#L211-L221

```bash
make docs
cd docs && make html SPHINXOPTS="-W -j 4"

Application error:
config directory doesn't contain a conf.py file (source)
make[1]: *** [html] Error 2
make: *** [docs] Error 2
```

transformers version: master branch

@sgugger

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR finishes to clean up some old references to sphinx in the setup or Makefile and updates the contributing guide/docs README to explain to users how to build the docs with our new tool or how to write them.

Fixes #14762

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -175,51 +175,82 @@ Follow these steps to start contributing:
 5. Develop the features on your branch.
 
    As you work on the features, you should make sure that the test suite
-   passes:
+   passes. You should run the tests impacted by your changes like this:
+
+   ```bash
+   $ pytest tests/test_to_run.py
+   ```
```
Comment (sgugger) [inline CONTRIBUTING.md:182]: I put this first as I think it's easier than just running `make test` (which I'm pretty sure no one does).

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -175,51 +175,82 @@ Follow these steps to start contributing:
...
+   For more information about tests, check out the
+   [dedicated documentation](https://huggingface.co/docs/transformers/testing)
+
+   🤗 Transformers relies on `black` and `isort` to format its source code
+   consistently. After you make changes, apply automatic style corrections and code verifications
+   that can't be automated in one go with:
 
    ```bash
-   $ python -m pytest -n 3 --dist=loadfile -s -v ./tests/
+   $ make fixup
```
Comment (sgugger) [inline CONTRIBUTING.md:200]: Put this first as it's more useful than `make style`.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -175,51 +175,82 @@ Follow these steps to start contributing:
 5. Develop the features on your branch.
 
    As you work on the features, you should make sure that the test suite
-   passes:
+   passes. You should run the tests impacted by your changes like this:
+
+   ```bash
+   $ pytest tests/test_to_run.py
```
Comment (LysandreJik) [inline CONTRIBUTING.md:181]: How about the following, to make sure we don't get issues about `ERROR: file or directory not found: tests/test_to_run.py`?

```suggestion
   $ pytest tests/<TEST_TO_RUN>.py
```

File: CONTRIBUTING.md (hunk 2 above)
Comment (LysandreJik) [reply CONTRIBUTING.md:200]: Yes, cool!

File: setup.py [hunk 4]
```diff
@@ -296,17 +287,9 @@ def run(self):
...
-    "sphinx",
-    "sphinx-markdown-tables",
-    "sphinx-rtd-theme",
-    "sphinx-copybutton",
-    "sphinxext-opengraph",
-    "sphinx-intl",
-)
+# Might need to add doc-builder and some specific deps in the future
+extras["docs_specific"] = []
+
```
Comment (LysandreJik) [inline setup.py:292]: This feels very good :)

**Review by LysandreJik — APPROVED** [review]
Looks good to me!

**Pull request merged**